doc/Task Hierarchy.md

```diff
@@ -0,0 +1,165 @@
+Task Type
+- Question Generation
+- Answer Generation: Reading Comprehension
+- Answer Generation: Contextual Question Answering
+- Answer Generation: Open book/Domain-specific Question Answering
+- Answer Generation: Commonsense Question Answering
+- Answer Generation: Fill in the Blank
+- Textual Entailment
+- Incorrect Answer Generation
+- Sentence Generation
+- Text Modification
+- Classification
+- Paragraph Generation
+- Verification
+- Translation
+- Multiple Choice Question Answering
+- Structured Text Generation
+- Hypernym Discovery
+- Parts-of-speech
+- Coreference: event coreference
+- Intent Detection
+- Claim Verification
+- Paraphrasing
+- Relation Prediction
+- Analogical Reasoning
+- Semantic Parsing
+- Stance Detection
+- Word Sense Disambiguation
+- Topic Generation
+- Mathematics: Arithmetic
+- Mathematics: Algebra
+- Mathematics: Geometry
+- Question Decomposition
+- Counting
+- Entity Detection
+- Role Labelling
+- Combinatorics
+- Grammar Error Correction
+- Summarization
+- Sentiment Analysis
+- Hate Speech Detection
+- Reasoning: Logic
+- Fake News Detection
+- Explanation Generation
+- Ethical Judgement
+- Style Transfer
+- Text Comparison
+- Question Composition
+- Review Generation
+- Column Matching
+- Handling Commands
+- Author Identification
+- Order Generation
+- Language Identification
+- Prompt Generation
+- Text Span Selection
+- Hallucination
+- Title Generation
+- Document Classification
+- Relevancy Estimation
+- Named Entity Recognition
+- Dialogue Understanding
+- Abuse Detection
+- Weblink Generation
+- Text Simplification
+- Reasoning: Numerical Reasoning
+- Coreference: Entity Coreference
+- Reasoning: Reasoning on Actions
+- Reasoning: General Commonsense Reasoning
+- Reasoning: Abductive Reasoning
+- Coreference: Pronoun Disambiguation
+- Reasoning: Temporal Reasoning
+- Reasoning: Spatial Reasoning
+- Reasoning: Multihop Reasoning
+- Reasoning: Reasoning on Social Interactions
+- Reasoning: Commonsense Reasoning
+- Reasoning: Numerical Commonsense Reasoning
+- Reasoning: Physical Reasoning
+- Reasoning: Logical Reasoning
+- Reasoning: Planning
+- Reasoning: Qualitative Reasoning
+- Reasoning: Deductive Reasoning
+- Reasoning: Causal Reasoning
+- Reasoning: Inverse-Causal Reasoning
```

Comment:
I would merge cause and inverse-causal. They seem to be using the same principles.